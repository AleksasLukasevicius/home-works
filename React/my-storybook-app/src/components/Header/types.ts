export interface HeaderProps {
  user?: User;
  onLogin: () => void;
  onLogout: () => void;
  onCreateAccount: () => void;
}

type User = {
  name: string;
};
