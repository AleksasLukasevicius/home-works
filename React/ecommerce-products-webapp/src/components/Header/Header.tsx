import { Logo } from "../Logo/Logo";
import { Link } from "react-router-dom";
import type { FC } from "react";
import { Grid, Typography } from "@mui/material";

export const Header: FC = () => {
  return (
    <Grid
      component="header"
      container
      spacing={1}
      mt={1}
      mb={1}
      alignItems="center"
    >
      <Grid item xs={12} sm={4}>
        <Link to="/">
          <Logo />
        </Link>
      </Grid>

      <Grid item xs={12} sm={4}>
        <Typography variant="h3">Alex Shop</Typography>
      </Grid>

      <Grid item xs={12} sm={4}>
        <Grid
          container
          component="nav"
          role="navigation"
          justifyContent="space-evenly"
          sx={{
            "& a": {
              // color: "purple",
              // ":hover": { color: "lightgreen" },
              textDecoration: "none",
            },
          }}
        >
          <Link to="/">
            <Typography aria-label="home link" variant="h6">
              Home
            </Typography>
          </Link>
          <Link to="/products">
            <Typography aria-label="products link" variant="h6">
              Products
            </Typography>
          </Link>
          <Link to="/cart">
            <Typography aria-label="cart link" variant="h6">
              Cart
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </Grid>
  );
};
