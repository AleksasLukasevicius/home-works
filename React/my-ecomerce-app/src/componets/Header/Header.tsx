import { Logo } from "../Logo/Logo";
import { Link } from "react-router-dom";

export const Header = () => {
  return (
    <header>
      <Link to="/">
        <Logo />
      </Link>
      <nav>
        <Link to="/products">Products</Link>
        <Link to="/cart">Cart</Link>
      </nav>
    </header>
  );
};
