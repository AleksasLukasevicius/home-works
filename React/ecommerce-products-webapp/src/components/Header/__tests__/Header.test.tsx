import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Header } from "../Header";
import renderer from "react-test-renderer";

describe("Header", () => {
  it.skip("should render Header naviagations links", () => {
    render(
      <BrowserRouter>
        (<Header />
        );
      </BrowserRouter>
    );

    expect(screen.getByLabelText("home link")).toBeVisible();
    expect(screen.getByLabelText("products link")).toBeVisible();
    expect(screen.getByLabelText("cart link")).toBeVisible();
  });

  it.each(["home link", "products link", "cart link"])(
    "should render Header naviagation %s",
    (link) => {
      render(
        <BrowserRouter>
          (<Header />
          );
        </BrowserRouter>
      );

      expect(screen.getByLabelText(link)).toBeVisible();
    }
  );

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          (<Header />
          );
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
