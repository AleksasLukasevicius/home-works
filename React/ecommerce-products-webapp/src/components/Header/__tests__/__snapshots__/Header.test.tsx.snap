// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
Array [
  "(",
  <header
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-10xf0xc-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="CodeA cademy logo"
          className="filter-green"
          src="logo.9725bb16.svg"
        />
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Alex Shop
      </h3>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
    >
      <nav
        className="MuiGrid-root MuiGrid-container css-1xixh9v-MuiGrid-root"
        role="navigation"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h6
            aria-label="home link"
            className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Home
          </h6>
        </a>
        <a
          href="/products"
          onClick={[Function]}
        >
          <h6
            aria-label="products link"
            className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Products
          </h6>
        </a>
        <a
          href="/cart"
          onClick={[Function]}
        >
          <h6
            aria-label="cart link"
            className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Cart
          </h6>
        </a>
      </nav>
    </div>
  </header>,
  ");",
]
`;
